apps/prairielearn/src/middlewares/autoEnroll.ts

```diff
@@ -8,13 +9,24 @@ export default asyncHandler(async (req, res, next) => {
   // they were enrolled, automatically enroll them. However, we will not
   // attempt to enroll them if they are an instructor (that is, if they have
   // a specific role in the course or course instance) or if they are
-  // impersonating another user.
+  // impersonating another user, or if self-enrollment requires a secret link.
```

Comment:
Nit: there's nothing in this code that deals with secret links yet, this feels misplaced? Should this be a `TODO` instead?

Reply:
Done in 898c52351fe5492b4061b339b3fd52c2f53b0df0.